AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -261,6 +260,42 @@ - (void)createWithPartition:(NSString *)partition
                                }];
 }
 
+- (void)deleteDocumentWithPartition:(NSString *)partition
+                         documentId:(NSString *)documentId
+                       writeOptions:(MSWriteOptions *)__unused writeOptions
+                  completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
+  // TODO consume writeOptions
+  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
+                   partitions:@[ partition ]
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+              // If error getting token.
+              if (error || !tokenResponses) {
+                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
+                completionHandler([[MSDataSourceError alloc] initWithError:error]);
+                return;
+              }
+
+              MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+              // Call CosmosDB
+              [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:cosmosDbIngestion
+                                                          tokenResult:tokenResponses.tokens[0]
+                                                           documentId:documentId
+                                                           httpMethod:@"DELETE"
+                                                                 body:[NSData data]
+                                                    completionHandler:^(NSData *__unused data, NSError *_Nonnull cosmosDbError) {
+                                                      // body returned from call (data) is empty
+                                                      NSNumber *errorCode = [cosmosDbError userInfo][kMSCosmosDbHttpCodeKey];
```

Comment:
Please consider rename variable `errorCode` into something related to status code, because not all status codes are invalid and it causes confusion when deletion succeeds.

Reply:
good point, made the name change.